Maria Cheprasova/1/parser.rb

```diff
@@ -0,0 +1,32 @@
+# fgkjl
+require 'nokogiri'
+require 'open-uri'
+require 'redis'
+
+# Hi there
+class Parser
+  URL = 'https://examples.yourdictionary.com/examples-of-puns.html'.freeze
+
+  def initialize
+    @wp = []
+    @db = Redis.new
+    parse
+    add_all_puns
+  end
+
+  def parse
+    doc = Nokogiri::HTML(URI.open(URL))
+    @wp = doc.css('#article_intro ul li').map(&:content)
+    @wp = doc.css('#article_body ul li').map(&:content)
```

Comment:
you can reverse and slice i this line:
.map(...).last(3)
also, avoid using modifying methods when possible